python/o80_pam/logger.py

```diff
@@ -0,0 +1,224 @@
+        self._segment_id = segment_id
+        self._frequency = frequency
+        self._id = str(id(self))
+        
+    def start(self):
+        """
+        starts the observations collecting process
+        """
+        
+        self._process = Process(target=_log,args=(self._segment_id,
+                                                  self._file_path,
+                                                  self._frequency,
+                                                  self._id))
+        _set_start(self._segment_id,self._id)
+        self._process.start()
+
+    def stop(self):
+        """
+        stops the observations collecting process
+        """
+        if hasattr(self,"_process"):
+            _set_stop(self._segment_id,self._id)
+            self._process.join()
+    
+    def __enter__(self):
+        """
+        For usage of this class as a context manager
+        """
+        self.start()
+        return self
+
+    def __exit__(self,_,__,___): 
+        """
+        For usage of this class as a context manager
+        """
+        self.stop()
+
+
+def read_file(file_path):
+    """
+    returns a generator of the observations stored in a file
+    created by an instance of Logger.
+    :param str file_path: path to the file to read
+    :raises :py:class:`FileNotFoundError`: if the the file does not exists
+    :returns: generator of instances of Observation 
+    """
+    if not os.path.isfile(file_path):
+        raise FileNotFoundError("failed to find {}".format(file_path))
+    # a serializer will take a string as input and generate a corresponding instance
+    # of observation
+    serializer = o80_pam.Serializer()
+    # size of a string representing an observation 
+    serialized_size = o80_pam.Serializer.serializable_size()
+    with open(file_path,"rb") as f:
+        # reading one serialized instance of observation 
+        instance_str = f.read(serialized_size)
+        while instance_str:
+            # deserializing it into an observation instance
+            observation = serializer.deserialize(instance_str)
+            # returing the instance
+            yield observation
+            instance_str = f.read(serialized_size)
+    # end of file
+    return
+
+
+class _File:
+
+    # convenience class for dealing with files of path
+    # /tmp/o80_pam_observations_num_date
+    # where num is a counter (or file id)
+    # and date is the date and time of the file creation
+    
+    def __init__(self,path,num,date):
+        self.path = path # absolute path to the file (name included)
+        self.num = num # num counter
+        self.date = date # date
+
+    @staticmethod
+    def create(prefix,path):
+        # given an path (i.e. /tmp/o80_pam_observations_num_date)
+        # and a prefix (o80_pam_observations)
+        # creates a corresponding instance of _File
+        # (i.e. extracts num and date from the path)
+        filename = path[path.find(prefix):]
```

Comment:
There might be some special cases, I'm missing but wouldn't do `os.path.basename(path)` do the same and be easier to understand?